createInvoice/src/main/java/com/cloudelements/demo/usecase/createinvoice/CreateInvoiceViewController.java

```diff
@@ -0,0 +1,196 @@
+package com.cloudelements.demo.usecase.createinvoice;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.http.client.ClientProtocolException;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.ParseException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import com.cloudelements.demo.model.ConfigurationField;
+import com.cloudelements.demo.model.Element;
+import com.cloudelements.demo.model.ElementConfiguration;
+import com.cloudelements.demo.model.ElementConfigurationOption;
+import com.cloudelements.demo.model.InvoiceTwo;
+import com.cloudelements.demo.model.Product;
+import com.cloudelements.demo.usecase.authentication.AuthenticationController;
+import com.cloudelements.demo.usecase.bulk.BulkController;
+import com.cloudelements.demo.usecase.element.ElementService;
+import com.cloudelements.demo.usecase.environment.EnvironmentService;
+import com.cloudelements.demo.util.AuthenticationUtil;
+import com.cloudelements.demo.util.HTTPUtil;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/*
+ * This is the main ViewController for the app
+ */
+@Controller
+public class CreateInvoiceViewController {
+
+	@Autowired
+	private EnvironmentService envService;
+	
+	@Autowired
+	private ElementService elService;
+	
+	@Autowired
+	private AuthenticationController authController;
+	
+	/*
+	 * Returns the first page of the invoice creation process - the authentication page
+	 * Notice it adds the elementKeys array onto the model, that's what is used in the UI to display the logos and logon details
+	 * 
+	 * Add any other element key to this array and refresh the page
+	 * 
+	 * The method also handles redirects once a CE instance has created and adds it onto the session so the UI can show
+	 */
+	@RequestMapping("/createInvoice")
+	public String init(Map<String, Object> model, HttpServletRequest request) throws ClientProtocolException, IOException {
+		if (!envService.cePropertiesAvailable()) {
+			model.put("error", "Please set your Cloud Elements environment variables before you continue. Go to <a href='/environment'>the environment page</a> to do so.");
+		}
+		
+		request.getSession().setAttribute("PAGETITLE", "Invoice creation use case");
+		
+		/* 
+		 * Add any element key from the catalog to this array to make it appear on the UI 
+		 * */
+		String[] elementKeys = { "netsuiteerpv2", "freshbooksv2", "quickbooks", "intacct", "sfdc" };
```

Comment:
Might be nice to put this in a config file or otherwise highlight the need to update it when adding a new element

Reply:
Comments added + to the readme / tutorial as well